spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/ConformanceExecution.scala

```diff
@@ -0,0 +1,173 @@
+                       (implicit spark: SparkSession, cmd: ConformanceConfig, dao: MenasDAO): DataFrame = {
+    val recordIdGenerationStrategy = getRecordIdGenerationStrategyFromConfig(conf)
+
+    implicit val featureSwitcher: FeatureSwitches = conformanceReader.readFeatureSwitches()
+
+    Try {
+      handleControlInfoValidation()
+      DynamicInterpreter.interpret(preparationResult.dataset, inputData)
+    } match {
+      case Failure(e: ValidationException) =>
+        AtumImplicits.SparkSessionWrapper(spark).setControlMeasurementError(sourceId.toString, e.getMessage, e.techDetails)
+        throw e
+      case Failure(NonFatal(e)) =>
+        val sw = new StringWriter
+        e.printStackTrace(new PrintWriter(sw))
+        AtumImplicits.SparkSessionWrapper(spark).setControlMeasurementError(sourceId.toString, e.getMessage, sw.toString)
+        throw e
+      case Success(conformedDF) =>
+        if (SchemaUtils.fieldExists(Constants.EnceladusRecordId, conformedDF.schema)) {
+          conformedDF // no new id regeneration
+        } else {
+          RecordIdGeneration.addRecordIdColumnByStrategy(conformedDF, Constants.EnceladusRecordId, recordIdGenerationStrategy)
+        }
+    }
+  }
+
+  protected def processConformanceResult(args: Array[String],
+                                         result: DataFrame,
+                                         preparationResult: PreparationResult,
+                                         menasCredentials: MenasCredentials)
+                                        (implicit spark: SparkSession,
+                                         cmd: ConformanceConfig,
+                                         fsUtils: FileSystemVersionUtils): Unit = {
+    val cmdLineArgs: String = args.mkString(" ")
+
+    PerformanceMetricTools.addJobInfoToAtumMetadata(
+      "conform",
+      preparationResult.pathCfg.inputPath,
+      preparationResult.pathCfg.outputPath,
+      menasCredentials.username, cmdLineArgs
+    )
+
+    val withPartCols = result
+      .withColumnIfDoesNotExist(InfoDateColumn, to_date(lit(cmd.reportDate), ReportDateFormat))
+      .withColumnIfDoesNotExist(InfoDateColumnString, lit(cmd.reportDate))
+      .withColumnIfDoesNotExist(InfoVersionColumn, lit(preparationResult.reportVersion))
+
+    val recordCount = result.lastCheckpointRowCount match {
+      case None => withPartCols.count
+      case Some(p) => p
+    }
+    if (recordCount == 0) {
+      handleEmptyOutput(SourcePhase.Conformance)
+    }
+
+    // ensure the whole path but version exists
+    fsUtils.createAllButLastSubDir(preparationResult.pathCfg.outputPath)
+
+    withPartCols.write.parquet(preparationResult.pathCfg.outputPath)
+
+    val publishDirSize = fsUtils.getDirectorySize(preparationResult.pathCfg.outputPath)
+    preparationResult.performance.finishMeasurement(publishDirSize, recordCount)
+    PerformanceMetricTools.addPerformanceMetricsToAtumMetadata(
+      spark,
+      "conform",
```

Comment:
let's keep it uniform
```suggestion
      "conformance",
```